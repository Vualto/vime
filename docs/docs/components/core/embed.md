---
title: vm-embed
sidebar_label: Embed
---

import Tabs from '@theme/Tabs'
import TabItem from '@theme/TabItem'

Embeds an external media player and enables interacting with it via `postMessage`. This is generally
used internally by other providers, but you could use it if your requirements are simple. You'll
also get the benefits of preconnections and lazy-loading. Refer to [existing providers](#used-by) to
see what params you can pass in, how to send commands to the player, and how to listen to events.

<!-- Auto Generated Below -->

## Usage

<Tabs
groupId="framework"
defaultValue="html"
values={[
{ label: 'HTML', value: 'html' },
{ label: 'React', value: 'react' },
{ label: 'Vue', value: 'vue' },
{ label: 'Svelte', value: 'svelte' },
{ label: 'Stencil', value: 'stencil' },
{ label: 'Angular', value: 'angular' }
]}>

<TabItem value="html">

```html
<vm-embed
  embed-src="https://www.youtube-nocookie.com/embed/DyTCOwB0DVw"
  params="autoplay=1&muted=1&controls=0"
  media-title="Agent 327: Operation Barbershop"
  origin="https://www.youtube-nocookie.com"
/>

<script>
  const embed = document.querySelector('vm-embed');

  embed.addEventListener('vmEmbedMessage', (e) => {
    const message = e.detail;
    // ...
  });
</script>
```

</TabItem>


<TabItem value="react">

```tsx {2,11-17}
import React from 'react';
import { Embed } from '@vime/react';

function Example() {
  const onMessage = (event: CustomEvent<any>) => {
    const message = event.detail;
    // ...
  };

  return (
    <Embed
      embedSrc="https://www.youtube-nocookie.com/embed/DyTCOwB0DVw"
      params={{ autoplay: 1, muted: 1, controls: 0 }}
      mediaTitle="Agent 327: Operation Barbershop"
      origin="https://www.youtube-nocookie.com"
      onVmEmbedMessage={onMessage}
    />
  );
}
```

</TabItem>


<TabItem value="vue">

```html {2-8,12,16} title="example.vue"
<template>
  <embed
    embedSrc="https://www.youtube-nocookie.com/embed/DyTCOwB0DVw"
    mediaTitle="Agent 327: Operation Barbershop"
    origin="https://www.youtube-nocookie.com"
    :params="params"
    @vmEmbedMessage="onMessage"
  />
</template>

<script>
  import { Embed } from '@vime/vue';

  export default {
    components: {
      Embed,
    },

    data: {
      params: {
        autoplay: 1,
        muted: 1,
        controls: 0,
      },
    },

    methods: {
      onMessage(message: any) {
        // ...
      },
    },
  };
</script>
```

</TabItem>


<TabItem value="svelte">

```tsx
<Embed
  embedSrc="https://www.youtube-nocookie.com/embed/DyTCOwB0DVw"
  origin="https://www.youtube-nocookie.com"
  mediaTitle="Agent 327: Operation Barbershop"
  params={params}
  on:vmEmbedMessage={onMessage}
/>
```

```html {2}
<script lang="ts">
  import { Embed } from '@vime/svelte';

  const params = {
    autoplay: 1,
    muted: 1,
    controls: 0,
  };

  const onMessage = (event: CustomEvent<any>) => {
    const message = event.detail;
    // ...
  };
</script>
```

</TabItem>


<TabItem value="stencil">

```tsx
class Example {
  private onMessage(event: CustomEvent<any>) {
    const message = event.detail;
    // ...
  }

  render() {
    return (
      <vm-embed
        embedSrc="https://www.youtube-nocookie.com/embed/DyTCOwB0DVw"
        params={{ autoplay: 1, muted: 1, controls: 0 }}
        mediaTitle="Agent 327: Operation Barbershop"
        origin="https://www.youtube-nocookie.com"
        onVmEmbedMessage={this.onMessage.bind(this)}
      />
    );
  }
}
```

</TabItem>


<TabItem value="angular">

```html title="example.html"
<vm-embed
  embed-src="https://www.youtube-nocookie.com/embed/DyTCOwB0DVw"
  media-title="Agent 327: Operation Barbershop"
  origin="https://www.youtube-nocookie.com"
  [params]="params"
  (vmEmbedMessage)="onMessage($event)"
/>
```

```ts title="example.ts"
class Example {
  params = {
    autoplay: 1,
    muted: 1,
    controls: 0,
  };

  onMessage(event: CustomEvent<any>) {
    const message = event.detail;
    // ...
  }
}
```

</TabItem>
    
</Tabs>


## Properties

| Property         | Attribute     | Description                                                                                                                                  | Type                                                              | Default     |
| ---------------- | ------------- | -------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------- | ----------- |
| `decoder`        | --            | A function which accepts the raw message received from the embedded media player via `postMessage` and converts it into a POJO.              | `((data: string) => Record<string, any> ∣ undefined) ∣ undefined` | `undefined` |
| `embedSrc`       | `embed-src`   | A URL that will load the external player and media (Eg: https://www.youtube.com/embed/DyTCOwB0DVw).                                          | `string`                                                          | `''`        |
| `mediaTitle`     | `media-title` | The title of the current media so it can be set on the inner `iframe` for screen readers.                                                    | `string`                                                          | `''`        |
| `origin`         | `origin`      | Where the src request had originated from without any path information.                                                                      | `string ∣ undefined`                                              | `undefined` |
| `params`         | `params`      | The parameters to pass to the embedded player which are appended to the `embedSrc` prop. These can be passed in as a query string or object. | `string ∣ { [x: string]: any; }`                                  | `''`        |
| `preconnections` | --            | A collection of URLs to that the browser should immediately start establishing a connection with.                                            | `string[]`                                                        | `[]`        |

## Events

| Event              | Description                                                                                                                                        | Type                  |
| ------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------- |
| `vmEmbedLoaded`    | Emitted when the embedded player and any new media has loaded.                                                                                     | `CustomEvent<void>`   |
| `vmEmbedMessage`   | Emitted when a new message is received from the embedded player via `postMessage`.                                                                 | `CustomEvent<any>`    |
| `vmEmbedSrcChange` | Emitted when the `embedSrc` or `params` props change. The payload contains the `params` serialized into a query string and appended to `embedSrc`. | `CustomEvent<string>` |

## Methods

### `postMessage(message: any, target?: string | undefined) => Promise<void>`

Posts a message to the embedded media player.

#### Returns

Type: `Promise<void>`

## Dependencies

### Used by

- [vm-dailymotion](../providers/dailymotion.md)
- [vm-vimeo](../providers/vimeo.md)
- [vm-youtube](../providers/youtube.md)

### Graph

```mermaid
graph TD;
  vm-dailymotion --> vm-embed
  vm-vimeo --> vm-embed
  vm-youtube --> vm-embed
  style vm-embed fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
