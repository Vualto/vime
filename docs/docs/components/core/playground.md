---
title: vm-playground
sidebar_label: Playground
---

A simple playground for testing and playing with Vime and its various providers.

<!-- Auto Generated Below -->

## Properties

| Property       | Attribute         | Description                                                         | Type                                                                                                                                            | Default                        |
| -------------- | ----------------- | ------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------ |
| `poster`       | `poster`          | The current poster to load.                                         | `string`                                                                                                                                        | `${BASE_MEDIA_URL}/poster.png` |
| `provider`     | `provider`        | The current media provider.                                         | `Provider.Audio ∣ Provider.Dailymotion ∣ Provider.Dash ∣ Provider.FakeTube ∣ Provider.HLS ∣ Provider.Video ∣ Provider.Vimeo ∣ Provider.YouTube` | `Provider.Audio`               |
| `showControls` | `show-controls`   | Whether to show the native controls or not.                         | `boolean`                                                                                                                                       | `true`                         |
| `showCustomUI` | `show-custom-u-i` | Whether to show the custom Vime UI or not.                          | `boolean`                                                                                                                                       | `false`                        |
| `src`          | `src`             | The current `src` to load into the provider.                        | `string ∣ undefined`                                                                                                                            | `undefined`                    |
| `theme`        | `theme`           | The current custom UI theme, won't work if custom UI is turned off. | `"dark" ∣ "light"`                                                                                                                              | `'dark'`                       |

## Dependencies

### Depends on

- [vm-audio](../providers/audio.md)
- [vm-video](../providers/video.md)
- [vm-hls](../providers/hls.md)
- [vm-dash](../providers/dash.md)
- [vm-youtube](../providers/youtube.md)
- [vm-vimeo](../providers/vimeo.md)
- [vm-dailymotion](../providers/dailymotion.md)
- [vm-player](player.md)
- [vm-default-ui](../ui/default-ui.md)

### Graph

```mermaid
graph TD;
  vm-playground --> vm-audio
  vm-playground --> vm-video
  vm-playground --> vm-hls
  vm-playground --> vm-dash
  vm-playground --> vm-youtube
  vm-playground --> vm-vimeo
  vm-playground --> vm-dailymotion
  vm-playground --> vm-player
  vm-playground --> vm-default-ui
  vm-audio --> vm-file
  vm-video --> vm-file
  vm-hls --> vm-video
  vm-dash --> vm-video
  vm-youtube --> vm-embed
  vm-vimeo --> vm-embed
  vm-dailymotion --> vm-embed
  vm-default-ui --> vm-ui
  vm-default-ui --> vm-icon-library
  vm-default-ui --> vm-skeleton
  vm-default-ui --> vm-click-to-play
  vm-default-ui --> vm-dbl-click-fullscreen
  vm-default-ui --> vm-captions
  vm-default-ui --> vm-poster
  vm-default-ui --> vm-spinner
  vm-default-ui --> vm-default-controls
  vm-default-ui --> vm-default-settings
  vm-default-controls --> vm-controls
  vm-default-controls --> vm-playback-control
  vm-default-controls --> vm-volume-control
  vm-default-controls --> vm-current-time
  vm-default-controls --> vm-control-spacer
  vm-default-controls --> vm-scrubber-control
  vm-default-controls --> vm-live-indicator
  vm-default-controls --> vm-end-time
  vm-default-controls --> vm-settings-control
  vm-default-controls --> vm-control-group
  vm-default-controls --> vm-fullscreen-control
  vm-default-controls --> vm-scrim
  vm-default-controls --> vm-caption-control
  vm-default-controls --> vm-time-progress
  vm-default-controls --> vm-pip-control
  vm-playback-control --> vm-control
  vm-playback-control --> vm-icon
  vm-playback-control --> vm-tooltip
  vm-volume-control --> vm-mute-control
  vm-volume-control --> vm-slider
  vm-mute-control --> vm-control
  vm-mute-control --> vm-icon
  vm-mute-control --> vm-tooltip
  vm-current-time --> vm-time
  vm-scrubber-control --> vm-slider
  vm-scrubber-control --> vm-tooltip
  vm-end-time --> vm-time
  vm-settings-control --> vm-control
  vm-settings-control --> vm-icon
  vm-settings-control --> vm-tooltip
  vm-fullscreen-control --> vm-control
  vm-fullscreen-control --> vm-icon
  vm-fullscreen-control --> vm-tooltip
  vm-time-progress --> vm-current-time
  vm-time-progress --> vm-end-time
  vm-pip-control --> vm-control
  vm-pip-control --> vm-icon
  vm-pip-control --> vm-tooltip
  vm-default-settings --> vm-menu-item
  vm-default-settings --> vm-menu-radio
  vm-default-settings --> vm-submenu
  vm-default-settings --> vm-menu-radio-group
  vm-default-settings --> vm-settings
  vm-menu-item --> vm-icon
  vm-menu-radio --> vm-menu-item
  vm-submenu --> vm-menu-item
  vm-submenu --> vm-menu
  vm-settings --> vm-menu
  style vm-playground fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
