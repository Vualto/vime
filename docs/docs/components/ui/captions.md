---
title: vm-captions
sidebar_label: Captions
---

import Tabs from '@theme/Tabs'
import TabItem from '@theme/TabItem'

Renders and displays VTT cues by hooking into the `textTracks` player property. This is a simple
implementation that can only handle rendering one text track, and one cue for the given track at a
time (even if many are active). The active track can be changed by setting the mode of any track
in the list to `showing`.

Be aware that after you set the text track mode to `showing`, the component will automatically set
it to hidden to avoid double captions. This also means that this component is **not recommended**
to be used in combination with the native HTML5 player controls.

## Visual

<img
  src="https://raw.githubusercontent.com/vime-js/vime/master/packages/core/src/components/ui/captions/captions.png"
  alt="Vime captions component"
/>

<!-- Auto Generated Below -->

## Usage

<Tabs
groupId="framework"
defaultValue="html"
values={[
{ label: 'HTML', value: 'html' },
{ label: 'React', value: 'react' },
{ label: 'Vue', value: 'vue' },
{ label: 'Svelte', value: 'svelte' },
{ label: 'Stencil', value: 'stencil' },
{ label: 'Angular', value: 'angular' }
]}>

<TabItem value="html">

```html {5}
<vm-player>
  <!-- ... -->
  <vm-ui>
    <!-- ... -->
    <vm-captions></vm-captions>
  </vm-ui>
</vm-player>
```

</TabItem>


<TabItem value="react">

```tsx {2,10}
import React from 'react';
import { Player, Ui, Captions } from '@vime/react';

function Example() {
  return (
    <Player>
      {/* ... */}
      <Ui>
        {/* ... */}
        <Captions />
      </Ui>
    </Player>
  );
}
```

</TabItem>


<TabItem value="vue">

```html {6,12,18} title="example.vue"
<template>
  <Player>
    <!-- ... -->
    <Ui>
      <!-- ... -->
      <Captions />
    </Ui>
  </Player>
</template>

<script>
  import { Player, Ui, Captions } from '@vime/vue';

  export default {
    components: {
      Player,
      Ui,
      Captions,
    },
  };
</script>
```

</TabItem>


<TabItem value="svelte">

```html {5,10} title="example.svelte"
<Player>
  <!-- ... -->
  <Ui>
    <!-- ... -->
    <Captions />
  </Ui>
</Player>

<script lang="ts">
  import { Player, Ui, Captions } from '@vime/svelte';
</script>
```

</TabItem>


<TabItem value="stencil">

```tsx {8}
class Example {
  render() {
    return (
      <vm-player>
        {/* ... */}
        <vm-ui>
          {/* ... */}
          <vm-captions />
        </vm-ui>
      </vm-player>
    );
  }
}
```

</TabItem>


<TabItem value="angular">

```html {5} title="example.html"
<vm-player>
  <!-- ... -->
  <vm-ui>
    <!-- ... -->
    <vm-captions></vm-captions>
  </vm-ui>
</vm-player>
```

</TabItem>
    
</Tabs>


## Properties

| Property         | Attribute         | Description                                                                                         | Type      | Default |
| ---------------- | ----------------- | --------------------------------------------------------------------------------------------------- | --------- | ------- |
| `controlsHeight` | `controls-height` | The height of any lower control bar in pixels so that the captions can reposition when it's active. | `number`  | `0`     |
| `hidden`         | `hidden`          | Whether the captions should be visible or not.                                                      | `boolean` | `false` |

## CSS Custom Properties

| Name                              | Description                                               |
| --------------------------------- | --------------------------------------------------------- |
| `--vm-captions-cue-bg-color`      | The background color of active cues.                      |
| `--vm-captions-cue-border-radius` | The border radius of active cues.                         |
| `--vm-captions-cue-padding`       | The padding around active cues.                           |
| `--vm-captions-font-size`         | Font size of captions when device screen size is <768px.  |
| `--vm-captions-font-size-large`   | Font size of captions when device screen size is >992px.  |
| `--vm-captions-font-size-medium`  | Font size of captions when device screen size is >768px.  |
| `--vm-captions-font-size-xlarge`  | Font size of captions when device screen size is >1200px. |
| `--vm-captions-text-color`        | The color of the captions text.                           |
| `--vm-captions-z-index`           | The position in the UI z-axis stack inside the player.    |

## Dependencies

### Used by

- [vm-default-ui](default-ui.md)

### Graph

```mermaid
graph TD;
  vm-default-ui --> vm-captions
  style vm-captions fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
