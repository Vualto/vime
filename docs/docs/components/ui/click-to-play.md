---
title: vm-click-to-play
sidebar_label: ClickToPlay
---

import Tabs from '@theme/Tabs'
import TabItem from '@theme/TabItem'

Enables toggling playback by clicking the player.

<!-- Auto Generated Below -->

## Usage

<Tabs
groupId="framework"
defaultValue="html"
values={[
{ label: 'HTML', value: 'html' },
{ label: 'React', value: 'react' },
{ label: 'Vue', value: 'vue' },
{ label: 'Svelte', value: 'svelte' },
{ label: 'Stencil', value: 'stencil' },
{ label: 'Angular', value: 'angular' }
]}>

<TabItem value="html">

```html {5}
<vm-player>
  <!-- ... -->
  <vm-ui>
    <!-- ... -->
    <vm-click-to-play></vm-click-to-play>
  </vm-ui>
</vm-player>
```

</TabItem>


<TabItem value="react">

```tsx {2,10}
import React from 'react';
import { Player, Ui, ClickToPlay } from '@vime/react';

function Example() {
  return (
    <Player>
      {/* ... */}
      <Ui>
        {/* ... */}
        <ClickToPlay />
      </Ui>
    </Player>
  );
}
```

</TabItem>


<TabItem value="vue">

```html {6,12,18} title="example.vue"
<template>
  <Player>
    <!-- ... -->
    <Ui>
      <!-- ... -->
      <ClickToPlay />
    </Ui>
  </Player>
</template>

<script>
  import { Player, Ui, ClickToPlay } from '@vime/vue';

  export default {
    components: {
      Player,
      Ui,
      ClickToPlay,
    },
  };
</script>
```

</TabItem>


<TabItem value="svelte">

```html {5,10} title="example.svelte"
<Player>
  <!-- ... -->
  <Ui>
    <!-- ... -->
    <ClickToPlay />
  </Ui>
</Player>

<script lang="ts">
  import { Player, Ui, ClickToPlay } from '@vime/svelte';
</script>
```

</TabItem>


<TabItem value="stencil">

```tsx {8}
class Example {
  render() {
    return (
      <vm-player>
        {/* ... */}
        <vm-ui>
          {/* ... */}
          <vm-click-to-play />
        </vm-ui>
      </vm-player>
    );
  }
}
```

</TabItem>


<TabItem value="angular">

```html {5} title="example.html"
<vm-player>
  <!-- ... -->
  <vm-ui>
    <!-- ... -->
    <vm-click-to-play></vm-click-to-play>
  </vm-ui>
</vm-player>
```

</TabItem>
    
</Tabs>


## Properties

| Property      | Attribute       | Description                                                                                            | Type      | Default |
| ------------- | --------------- | ------------------------------------------------------------------------------------------------------ | --------- | ------- |
| `useOnMobile` | `use-on-mobile` | By default this is disabled on mobile to not interfere with playback, set this to `true` to enable it. | `boolean` | `false` |

## CSS Custom Properties

| Name                         | Description                                            |
| ---------------------------- | ------------------------------------------------------ |
| `--vm-click-to-play-z-index` | The position in the UI z-axis stack inside the player. |

## Dependencies

### Used by

- [vm-default-ui](default-ui.md)

### Graph

```mermaid
graph TD;
  vm-default-ui --> vm-click-to-play
  style vm-click-to-play fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
