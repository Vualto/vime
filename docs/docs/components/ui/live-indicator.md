---
title: vm-live-indicator
sidebar_label: LiveIndicator
---

import Tabs from '@theme/Tabs'
import TabItem from '@theme/TabItem'

This can be used to indicate to the user that the current media is being streamed live.

## Visual

<img
  src="https://raw.githubusercontent.com/vime-js/vime/master/packages/core/src/components/ui/live-indicator/live-indicator.png"
  alt="Vime live indicator component"
/>

<!-- Auto Generated Below -->

## Usage

<Tabs
groupId="framework"
defaultValue="html"
values={[
{ label: 'HTML', value: 'html' },
{ label: 'React', value: 'react' },
{ label: 'Vue', value: 'vue' },
{ label: 'Svelte', value: 'svelte' },
{ label: 'Stencil', value: 'stencil' },
{ label: 'Angular', value: 'angular' }
]}>

<TabItem value="html">

```html {7}
<vm-player>
  <!-- ... -->
  <vm-ui>
    <!-- ... -->
    <vm-controls>
      <!-- ... -->
      <vm-live-indicator></vm-live-indicator>
    </vm-controls>
  </vm-ui>
</vm-player>
```

</TabItem>


<TabItem value="react">

```tsx {6,16}
import React from 'react';
import { Player, Ui, Controls, LiveIndicator } from '@vime/react';

function Example() {
  return (
    <Player>
      {/* ... */}
      <Ui>
        {/* ... */}
        <Controls>
          <LiveIndicator />
        </Controls>
      </Ui>
    </Player>
  );
}
```

</TabItem>


<TabItem value="vue">

```html {7,18,26} title="example.vue"
<template>
  <Player>
    <!-- ... -->
    <Ui>
      <!-- ... -->
      <Controls>
        <LiveIndicator />
      </Controls>
    </Ui>
  </Player>
</template>

<script>
  import { Player, Ui, Controls, LiveIndicator } from '@vime/vue';

  export default {
    components: {
      Player,
      Ui,
      Controls,
      LiveIndicator,
    },
  };
</script>
```

</TabItem>


<TabItem value="svelte">

```html {6,17} title="example.svelte"
<Player>
  <!-- ... -->
  <Ui>
    <!-- ... -->
    <Controls>
      <LiveIndicator />
    </Controls>
  </Ui>
</Player>

<script lang="ts">
  import { Player, Ui, Icons, Controls, LiveIndicator } from '@vime/svelte';
</script>
```

</TabItem>


<TabItem value="stencil">

```tsx {9}
class Example {
  render() {
    return (
      <vm-player>
        {/* ... */}
        <vm-ui>
          {/* ... */}
          <vm-controls>
            <vm-live-indicator />
          </vm-controls>
        </vm-ui>
      </vm-player>
    );
  }
}
```

</TabItem>


<TabItem value="angular">

```html {7} title="example.html"
<vm-player>
  <!-- ... -->
  <vm-ui>
    <!-- ... -->
    <vm-controls>
      <!-- ... -->
      <vm-live-indicator></vm-live-indicator>
    </vm-controls>
  </vm-ui>
</vm-player>
```

</TabItem>
    
</Tabs>


## CSS Custom Properties

| Name                        | Description                          |
| --------------------------- | ------------------------------------ |
| `--vm-live-indicator-color` | The color of the live indicator dot. |

## Dependencies

### Used by

- [vm-default-controls](controls/default-controls.md)

### Graph

```mermaid
graph TD;
  vm-default-controls --> vm-live-indicator
  style vm-live-indicator fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
