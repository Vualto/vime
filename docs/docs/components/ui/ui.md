---
title: vm-ui
sidebar_label: UI
---

import Tabs from '@theme/Tabs'
import TabItem from '@theme/TabItem'

Simple container that holds a collection of user interface components.

The only important role this component really has is, avoiding overlapping custom UI with the
native iOS media player UI. Therefore, custom UI is only displayed on iOS if the `playsinline` prop
is `true`, and the player is not in fullscreen mode.

<!-- Auto Generated Below -->

## Usage

<Tabs
groupId="framework"
defaultValue="html"
values={[
{ label: 'HTML', value: 'html' },
{ label: 'React', value: 'react' },
{ label: 'Vue', value: 'vue' },
{ label: 'Svelte', value: 'svelte' },
{ label: 'Stencil', value: 'stencil' },
{ label: 'Angular', value: 'angular' }
]}>

<TabItem value="html">

```html {3-5}
<vm-player>
  <!-- ... -->
  <vm-ui>
    <!-- ... -->
  </vm-ui>
</vm-player>
```

</TabItem>


<TabItem value="react">

```tsx {2,8}
import React from 'react';
import { Player, Ui } from '@vime/react';

function Example() {
  return (
    <Player>
      {/* ... */}
      <Ui>{/* ... */}</Ui>
    </Player>
  );
}
```

</TabItem>


<TabItem value="vue">

```html {4-6,11,16} title="example.vue"
<template>
  <Player>
    <!-- ... -->
    <Ui>
      <!-- ... -->
    </Ui>
  </Player>
</template>

<script>
  import { Player, Ui } from '@vime/vue';

  export default {
    components: {
      Player,
      Ui,
    },
  };
</script>
```

</TabItem>


<TabItem value="svelte">

```html {3-5,9} title="example.svelte"
<Player>
  <!-- ... -->
  <Ui>
    <!-- ... -->
  </Ui>
</Player>

<script lang="ts">
  import { Player, Ui } from '@vime/svelte';
</script>
```

</TabItem>


<TabItem value="stencil">

```tsx {6}
class Example {
  render() {
    return (
      <vm-player>
        {/* ... */}
        <vm-ui>{/* ... */}</vm-ui>
      </vm-player>
    );
  }
}
```

</TabItem>


<TabItem value="angular">

```html {3-5} title="example.html"
<vm-player>
  <!-- ... -->
  <vm-ui>
    <!-- ... -->
  </vm-ui>
</vm-player>
```

</TabItem>
    
</Tabs>


## Slots

| Slot | Description                                   |
| ---- | --------------------------------------------- |
|      | Used to pass in UI components for the player. |

## CSS Custom Properties

| Name              | Description                                              |
| ----------------- | -------------------------------------------------------- |
| `--vm-ui-z-index` | The position in the root z-axis stack inside the player. |

## Dependencies

### Used by

- [vm-default-ui](default-ui.md)

### Graph

```mermaid
graph TD;
  vm-default-ui --> vm-ui
  style vm-ui fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
