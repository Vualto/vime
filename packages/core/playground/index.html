<!DOCTYPE html>
<html dir="ltr">

<head>
  <meta charset="UTF-8">
  <title>Playground | Vime</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">

  <link rel="stylesheet" href="/src/themes/default.css">
  <link rel="stylesheet" href="/src/themes/light.css">
  <link rel="icon" type="image/png" href="/playground/favicon.ico">
  <script type="module" src="/dist/vime/vime.esm.js"></script>

  <style>
    html, body {
      width: 100%;
      height: 100%;
      margin: 0;
      box-sizing: border-box;
    }

    body {
      padding: 16px;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    }
  </style>
</head>
<body>
  <vm-playground></vm-playground>
</body>
</html>
