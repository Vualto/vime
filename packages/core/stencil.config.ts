import { Config } from '@stencil/core';
import { svelteOutputTarget } from './targets/svelte';
import { reactOutputTarget } from './targets/react';
import { angularOutputTarget } from './targets/angular';
import { vueOutputTarget } from './targets/vue';
import { vueNextOutputTarget } from './targets/vue-next';

export const config: Config = {
  namespace: 'Vime',
  taskQueue: 'async',
  outputTargets: [
    reactOutputTarget(),
    angularOutputTarget(),
    vueOutputTarget(),
    vueNextOutputTarget(),
    svelteOutputTarget(),
    {
      type: 'dist',
      esmLoaderPath: '../loader',
    },
    {
      type: 'dist-custom-elements-bundle',
    },
  ],
  testing: {
    globals: {
      'ts-jest': {
        diagnostics: false,
      },
    },
    setupFilesAfterEnv: [
      './src/globals/jest.js',
    ],
    coverageDirectory: 'jest-coverage',
    // @ts-ignore
    watchPlugins: [
      'jest-watch-typeahead/filename',
      'jest-watch-typeahead/testname',
    ],
    testPathIgnorePatterns: [
      '/node_modules/',
      '/cypress/',
    ],
  },
  preamble: '(C) Vime https://vimejs.com - MIT License',
};
