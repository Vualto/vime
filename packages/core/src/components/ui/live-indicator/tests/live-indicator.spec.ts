import { SpecPage } from '@stencil/core/testing';
import { newUISpecPage } from '../../ui/tests';
import { LiveIndicator } from '../live-indicator';

let page: SpecPage;
let indicator: HTMLVmLiveIndicatorElement;

beforeEach(async () => {
  ({ page } = await newUISpecPage(
    [LiveIndicator],
    '<vm-live-indicator />',
  ));

  indicator = page.root!.querySelector('vm-live-indicator')!;
});

it('should be structurally sound', () => {
  expect(indicator).toMatchSnapshot();
});

it('should contain text', () => {
  expect(indicator.innerHTML).toContain('LIVE');
});

it('should be hidden by default', () => {
  expect(indicator).toHaveClass('hidden');
});

it('should not be hidden if media is live', async () => {
  expect(indicator).toHaveClass('hidden');
  indicator.isLive = true;
  await page.waitForChanges();
  expect(indicator).not.toHaveClass('hidden');
});
