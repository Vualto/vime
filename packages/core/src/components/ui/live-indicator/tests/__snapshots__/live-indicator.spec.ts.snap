// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be structurally sound 1`] = `
<vm-live-indicator class="hidden">
  <div class="indicator"></div>
  LIVE
</vm-live-indicator>
`;
